### PR TITLE
Tell player why turn is invalid

### PR DESCRIPTION
**Pull request by blaiseIT**

### CODE REVIEW

File: src/roomObj.js [hunk 1]
```diff
@@ -104,12 +106,9 @@ exports.Room = () => ({
...
 
     gameApplyTurn() {
-        let points;
-        /* Validate the whole turn by passing the array of moves to a method
-         * If the turn is valid, end the player's turn
-         * Else, return an error and undo the partial turn */
-        if (this.boardBackups[0].isTurnValid(this.moves)) {
-            points = this.board.isGameWon();
+        // Frontend should ensure this function is only called if the turn is valid
+        if (this.board.turnValidity > 0) {
```
Comment (blaiseIT) [inline src/roomObj.js:110]: Maybe make this comparison more explicit

File: src/gameUtil.js [hunk 2]
```diff
@@ -29,6 +37,9 @@ exports.Board = () => ({
     diceRolled: new Array(2),
     dice: new Array(2),
     recentMove: {},
+    possibleTurns: null,
```
Comment (blaiseIT) [inline src/gameUtil.js:40]: Do we want this big object to be on the board and shared with the frontend?

Comment (michaelti) [reply src/gameUtil.js:40]: Probably not unless we find a useful purpose for it. One solution could be a util function to filter the board down to its "public" properties before we broadcast it

File: src/roomObj.js (hunk 1 above)
Comment (michaelti) [reply src/roomObj.js:110]: 🆗

File: src/gameUtil.js [hunk 3]
```diff
@@ -29,6 +37,26 @@ exports.Board = () => ({
...
+    maxTurnLength: 0,
+    turnValidity: TurnMessage.invalid,
+
+    publicProperties() {
```
Comment (blaiseIT) [inline src/gameUtil.js:44]: I would like this to be a getter (i.e`get publicProperties()`) but that doesn't seem to work

Comment (michaelti) [reply src/gameUtil.js:44]: IIRC we'd need to be using ES6 classes for that

**Review by michaelti — APPROVED** [review]

**Pull request merged**